### PR TITLE
Ruby style guide

### PR DESCRIPTION
**Pull request by jaredbeck**
I've been happy on other projects using a linter for ruby style, even making the linter part of the build process. For example, a linter can nitpick PRs in a way that would be annoying coming from me. The trick, as usual where code style is concerned, is agreeing on a style :) I like most of [bbatsov/ruby-style-guide][1], and the associated linter [bbatsov/rubocop][2].  What do you like, Ben? @batter 

[1]: https://github.com/bbatsov/ruby-style-guide
[2]: https://github.com/bbatsov/rubocop

### CODE REVIEW

**Comment by batter** [comment]
Hey, I don't have much experience working with a strict linter other than some basic Javascript linting within a webpack compiled project, but I do realize that the library could be up to slightly better code standards so I don't have an objection with a linter.  I'd like to examine this a little more closely when I have some time later in the week.

File: lib/paper_trail.rb [hunk 1]
```diff
@@ -101,7 +101,8 @@ def self.serializer
   end
 
   def self.active_record_protected_attributes?
-    @active_record_protected_attributes ||= ::ActiveRecord::VERSION::MAJOR < 4 || !!defined?(ProtectedAttributes)
+    @active_record_protected_attributes ||= ::ActiveRecord::VERSION::MAJOR < 4 ||
+      !defined?(ProtectedAttributes).nil?
```
Comment (batter) [inline lib/paper_trail.rb]: I actually think `!!defined?(ProtectedAttributes)` is clearer than the double negative of `!defined?(ProtectedAttributes).nil?`

Comment (jaredbeck) [inline lib/paper_trail.rb]: OK, I reverted that change.

**Comment by jaredbeck** [comment]
Other than the double-negative (https://github.com/airblade/paper_trail/pull/643#discussion_r43921922), did you have any concerns? Mind if I merge?

Note that this PR just adds `rubocop` as a dev. dependency, but doesn't introduce any linting to the build process yet. So, this PR just makes a linter available to use, but not required.

**Comment by batter** [comment]
Seems reasonable to me, at least gives us the option. I know that the style and code cleanliness has been lacking within the project over the years so anything we can do to help improve code quality and modularity is good in my book.

**Pull request merged**